### ISSUE TITLE
Add Llama 3 8B to TT-Buda model demos

### ISSUE THREAD

**Issue by EwoutH** [tenstorrent/tt-buda-demos#59]
Meta's [Llama 3](https://llama.meta.com/llama3/) is currently the leading edge of <10B models and is highly rated on the user-preference based Chatbot Arena. Being able to run inference of such a powerful model could greatly increase the value and percieved potential of Tenstorrent's Grayskull accelerators.

Quantization is needed to fit an 8B model on the 8GB LPDDR4 of the Grayskull cards. This might be possible with the block floating point format Grayskull supports, specifically BFP4. This would make it probably more complicated than adding other models.

Given the complexity and the benefits that enabling inference of such a SOTA model would be, prioritising it internally or putting a significant bounty on it could be considered.

**Comment by milank94** [comment]
Great feedback @EwoutH! 

CC: @Shubhamsaboo @staylorTT @adifrancescoTT -- great suggestion: prioritizing it internally or putting a significant bounty on it could be considered.

**Comment by Shubhamsaboo** [comment]
Will keep that in mind, thanks @EwoutH!

**Comment by EwoutH** [comment]
AMD [launched](https://www.anandtech.com/show/21419/amd-announces-the-ryzen-ai-300-series-for-mobile-zen-5-with-rdna-35-and-xdna2-npu-with-50-tops) their Ryzen AI 300 series APUs today, with Block FP16 support in their NPUs. That means block floating point might gain in adoption, and Tenstorrent could be on the leading edge of that if they demo BFP4 use in their Grayskull cards.

**Comment by EwoutH** [comment]
Congratulations on the [release](https://tenstorrent.com/vision/tenstorrent-launches-next-generation-wormhole-based-developer-kits-and-workstations) of the [Wormhole](https://tenstorrent.com/hardware/wormhole) n150s and n300s! With 12GB memory and 288 GB/s bandwidth, it should allow running 8-bit quants of LLama 3 8B at over 30 tokens per second (memory wise).

**Comment by milank94** [comment]
Thanks @EwoutH! Yes, Llama-3 8B bring-up on n150 is underway.

**Comment by staylorTT** [comment]
@nvukobratTT Lets see if this is something MCW can bring up.

**Comment by milank94** [comment]
Llama3.1-8B is being worked on through the TT-Metalium stack. I encourage you to track progress here: https://github.com/tenstorrent/tt-metal

Llama3.2-11B is also being brought up on the n300 device.

Moving forward, TT-Buda support for new model bring up and optimizations will be limited as we transition to the TT-Forge stack.

**Comment by EwoutH** [comment]
> Moving forward, TT-Buda support for new model bring up and optimizations will be limited as we transition to the TT-Forge stack.

@milank94 is this still the current direction? If so, it might be good to communicate about that wider, like adding a notice at the top of the tt-budas Readme and documentation.

**Comment by milank94** [comment]
That's correct, @EwoutH. Good suggestion. We'll add a notice to the repos and docs to reflect this.

**Comment by milank94** [comment]
We've added banners now to the major TT-Buda repos with this notice
* https://github.com/tenstorrent/tt-buda-demos
* https://github.com/tenstorrent/tt-buda
* https://github.com/tenstorrent/tt-budabackend

**Comment by EwoutH** [comment]
Thanks, appreciated!
